src/aatoolbox/datasources/fewsnet/fewsnet.py

```
@@ -27,243 +32,218 @@
-            output_dir=output_dir_region,
+        # FN data can be regional, and thus we save the raw data
+        # in "glb" instead of in the iso3 folder, even though part of it is at
+        # country level
+        self._raw_base_dir = self._get_base_dir(
+            is_public=True,
+            is_raw=True,
+            is_global=True,
         )
-    else:
-        region_data = output_dir_region
-    return region_data
 
+        self._iso2 = Country.get_iso2_from_iso3(self._country_config.iso3)
+        if self._iso2 is None:
+            logger.error("No ISO2 found for the given ISO3. Check your ISO3.")
+
+    def download(
+        self,
+        date_pub: Union[date, str],
+        clobber: bool = False,
+    ) -> Path:
+        """
+        Retrieve the raw fewsnet data.
+
+        Depending on the region and date, this data is published per region or
+        per country. This function retrieves the country data
+        if it exists, and else the regional data for `date_pub`.
+
+        Parameters
+        ----------
+        date_pub : date or str
+            date for which the data should be downloaded
+            only the year and month part are used
+            this commonly refers to the month of the Current Situation period
+            if str, it should be in ISO 8601 format, e.g. '2021-10-01'
+        clobber : bool, default = False
+            If True, overwrites existing raw files
+
+        Examples
+        --------
+        >>> download(date_pub="2020-10-01")
+        """
+        # convert to datetime if str
+        if not isinstance(date_pub, date):
+            date_pub = datetime.strptime(date_pub, _DATE_FORMAT)
+
+        # we prefer the country data as this more nicely structured
+        # thus first check if that is available
+        country_path = self._download_country(
+            date_pub=date_pub,
+            clobber=clobber,
+        )
+        if country_path is not None:
```

I think it would be better if you raised an error in `_download_zip` and let it bubble up, instead passing `None` all the way back here. Then instead of all this logic here you can use try-catch.

---

I know I say this almost every PR, but thanks for making this code so much better!